jbmc/src/java_bytecode/java_static_initializers.cpp

```diff
@@ -209,6 +220,43 @@ static void clinit_wrapper_do_recursive_calls(
     code_function_callt call_real_init;
     call_real_init.function() = find_sym_it->second.symbol_expr();
     init_body.move_to_operands(call_real_init);
+
+    // Add a standard nondet initialisation for each global declared on this
+    // class. Note this is the same invocation used in
+    // get_stub_initializer_body and java_static_lifetime_init.
+    if(nondet_static)
+    {
+      object_factory_parameterst parameters = object_factory_parameters;
+      parameters.function_id = clinit_wrapper_name(class_name);
+
+      for(const auto &symbol : symbol_table.symbols)
+      {
+        if(
+          symbol.second.type.get(ID_C_class) == class_name &&
+          symbol.second.is_static_lifetime &&
+          !symbol.second.type.get_bool(ID_C_constant))
+        {
+          const symbol_exprt new_global_symbol = symbol.second.symbol_expr();
+
+          parameters.max_nonnull_tree_depth =
+            is_non_null_library_global(symbol.first)
+              ? std::max(
+                  size_t(1), object_factory_parameters.max_nonnull_tree_depth)
+              : object_factory_parameters.max_nonnull_tree_depth;
+
+          gen_nondet_init(
```

Comment:
Careful -- gen-nondet-init mutates the symbol table, which invalidates iterators over the symbol table. You must first pick which identifiers to mess with, then retrieve and run gen-nondet-init for each in turn.

Reply:
Done :)